Specs/Scene/ModelExperimental/SkinningPipelineStageSpec.js

```diff
@@ -0,0 +1,183 @@
+import ShaderBuilderTester from "../../ShaderBuilderTester.js";
+
+describe(
+  "Scene/ModelExperimental/SkinningPipelineStage",
+  function () {
+    let scene;
+    const gltfLoaders = [];
+
+    beforeAll(function () {
+      scene = createScene();
+    });
+
+    afterAll(function () {
+      scene.destroyForSpecs();
+    });
+
+    afterEach(function () {
+      const gltfLoadersLength = gltfLoaders.length;
+      for (let i = 0; i < gltfLoadersLength; ++i) {
+        const gltfLoader = gltfLoaders[i];
+        if (!gltfLoader.isDestroyed()) {
+          gltfLoader.destroy();
+        }
+      }
+      gltfLoaders.length = 0;
+      ResourceCache.clearForSpecs();
+    });
+
+    function getOptions(gltfPath, options) {
+      const resource = new Resource({
+        url: gltfPath,
+      });
+
+      return combine(options, {
+        gltfResource: resource,
+        incrementallyLoadTextures: false, // Default to false if not supplied
+      });
+    }
+
+    function loadGltf(gltfPath, options) {
+      const gltfLoader = new GltfLoader(getOptions(gltfPath, options));
+      gltfLoaders.push(gltfLoader);
+      gltfLoader.load();
+
+      return waitForLoaderProcess(gltfLoader, scene);
+    }
+
+    const simpleSkinUrl =
+      "./Data/Models/GltfLoader/SimpleSkin/glTF/SimpleSkin.gltf";
+    const cesiumManUrl =
+      "./Data/Models/DracoCompression/CesiumMan/CesiumMan.gltf";
+
+    it("processes skin with two joints", function () {
+      const mockJointMatrices = [
+        new Matrix4(),
+        Matrix4.clone(Matrix4.IDENTITY),
+      ];
+      const renderResources = {
+        attributes: [],
+        shaderBuilder: new ShaderBuilder(),
+        attributeIndex: 1,
+        model: {
+          type: ModelExperimentalType.TILE_GLTF,
+        },
+        runtimeNode: {
+          computedJointMatrices: mockJointMatrices,
+        },
+      };
+
+      return loadGltf(simpleSkinUrl).then(function (gltfLoader) {
+        const components = gltfLoader.components;
+        const primitive = components.nodes[0].primitives[0];
+
+        SkinningPipelineStage.process(renderResources, primitive);
+
+        const shaderBuilder = renderResources.shaderBuilder;
+
+        ShaderBuilderTester.expectHasVertexDefines(shaderBuilder, [
+          "HAS_SKINNING",
+        ]);
+
+        ShaderBuilderTester.expectHasVertexFunction(
+          shaderBuilder,
+          SkinningPipelineStage.FUNCTION_ID_GET_SKINNING_MATRIX,
+          SkinningPipelineStage.FUNCTION_SIGNATURE_GET_SKINNING_MATRIX,
+          [
+            "    mat4 skinnedMatrix = mat4(0);",
+            "    skinnedMatrix += a_weights_0.x * u_jointMatrices[int(a_joints_0.x)];",
+            "    skinnedMatrix += a_weights_0.y * u_jointMatrices[int(a_joints_0.y)];",
+            "    skinnedMatrix += a_weights_0.z * u_jointMatrices[int(a_joints_0.z)];",
+            "    skinnedMatrix += a_weights_0.w * u_jointMatrices[int(a_joints_0.w)];",
```

Comment:
Just to make sure I read the spec correctly, even though there's two joints, this still works because `a_weights_0.z`, `a_weights_0.w` would both be 0 in the glTF, correct?

Reply:
Yes. The [Skins](https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_020_Skins.md) section of the glTF Tutorial repo might help illustrate this since it breaks down an actual example.